src/framework/pool_hash.c

```diff
@@ -1,21 +1,25 @@
-#include <stdlib.h>
+#define NULL_CHARACTER '\0'
 
-#ifdef UNDERSCORE
-#define pool_hash pool_hash_
-#else
-#ifdef DOUBLEUNDERSCORE
-#define pool_hash pool_hash__
-#endif
-#endif
+/*
+ use iso_c_binding, only : c_int, c_char
 
-void pool_hash(int* hash, char* key, int* len)
+ interface
+   subroutine pool_hash(hash, key, len) bind(c)
+      use iso_c_binding, only : c_int, c_char
+      integer (c_int), intent(inout) :: hash
+      character (c_char), intent(in) :: key
```

Comment:
I think `key` should also be declared as a character array.

Reply:
Also sounds good. I updated this reference to the interface here, and updated the interface within `mpas_pool_routines.F` (lines ~6000).